server/src/assignment/assignment.service.ts

```diff
@@ -0,0 +1,79 @@
+import { Injectable } from '@nestjs/common';
+import { Repository, UpdateResult } from 'typeorm';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Guest } from '@entities/guest.entity';
+import { Assignment } from '@entities/assignment.entity';
+
+import { assignmentData, primaryData } from '@libs/person';
+
+@Injectable()
+export class AssignmentService {
+	constructor(
+		@InjectRepository(Assignment)
+		private assignmentRepository: Repository<Assignment>,
+	) {}
+
+	/**
+	 * Add assignments for a guest.
+	 * @param guest The guest to assign roles to
+	 * @param assignment Wedding assignment information
+	 * @returns The assignments given to the guest
+	 */
+	async create(
+		guest: primaryData,
+		assignment: assignmentData,
+	): Promise<assignmentData> {
+		return !assignment
+			? null
+			: await this.assignmentRepository.save({
+					...assignment,
+					guest,
+			  });
+	}
+
+	/**
+	 * Update a guest's assigned information
+	 * @param uuid The uuid of the guest to update
+	 * @param assignmentUpdate The new assignments for the guest
+	 * @returns The result of the update
+	 */
+	async update(
+		uuid: string,
+		assignmentUpdate: assignmentData,
+	): Promise<UpdateResult> {
+		const assignmentUpdateData = {
+			table: assignmentUpdate.table,
+			isInGroomishParty: assignmentUpdate.isInGroomishParty,
+			isInBridalParty: assignmentUpdate.isInBridalParty,
+			isFamily: assignmentUpdate.isFamily,
+			pokemon: assignmentUpdate.pokemon,
+		};
+
+		if (!Object.values(assignmentUpdateData).every((el) => el == null))
```

Comment:
Very small point of order - why not do this check on assignmentUpdate, then only create assignmentUpdateData if there is something to update?

Like I said, very small.